cmd/kueuectl/app/delete/delete_workload_test.go

```diff
@@ -0,0 +1,429 @@
+		wantOutErr    string
+		wantErr       string
+	}{
+		"shouldn't delete a workload because no arguments": {
+			args:    []string{},
+			wantErr: "requires at least 1 arg(s), only received 0",
+		},
+		"shouldn't delete a workload because not confirmed": {
+			args: []string{"wl1"},
+			workloads: []runtime.Object{
+				testingworkload.MakeWorkload("wl1", metav1.NamespaceDefault).OwnerReference(jobGVK, "j1", "").Obj(),
+				testingworkload.MakeWorkload("wl2", metav1.NamespaceDefault).Obj(),
+			},
+			jobs: []runtime.Object{
+				&bactchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: "j1", Namespace: metav1.NamespaceDefault}},
+				&bactchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: "j2", Namespace: metav1.NamespaceDefault}},
+			},
+			gvk: schema.GroupVersionKind{Group: "batch", Version: "v1", Kind: "Job"},
+			wantWorkloads: []v1beta1.Workload{
+				*testingworkload.MakeWorkload("wl1", metav1.NamespaceDefault).OwnerReference(jobGVK, "j1", "").Obj(),
+				*testingworkload.MakeWorkload("wl2", metav1.NamespaceDefault).Obj(),
+			},
+			wantJobList: &bactchv1.JobList{
+				TypeMeta: metav1.TypeMeta{Kind: "JobList", APIVersion: "batch/v1"},
+				Items: []bactchv1.Job{
+					{
+						TypeMeta:   metav1.TypeMeta{Kind: "Job", APIVersion: "batch/v1"},
+						ObjectMeta: metav1.ObjectMeta{Name: "j1", Namespace: metav1.NamespaceDefault},
+					},
+					{
+						TypeMeta:   metav1.TypeMeta{Kind: "Job", APIVersion: "batch/v1"},
+						ObjectMeta: metav1.ObjectMeta{Name: "j2", Namespace: metav1.NamespaceDefault},
+					},
+				},
+			},
+			wantOut: `This operation will also delete:
+  - jobs.batch/j1 associated with the wl1 workload
+Do you want to proceed (y/n)? Deletion is canceled
+`,
+		},
+		"should delete a workload without confirmation because no associated job": {
+			args: []string{"wl2"},
+			workloads: []runtime.Object{
+				testingworkload.MakeWorkload("wl1", metav1.NamespaceDefault).OwnerReference(jobGVK, "j1", "").Obj(),
+				testingworkload.MakeWorkload("wl2", metav1.NamespaceDefault).Obj(),
+			},
+			jobs: []runtime.Object{
+				&bactchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: "j1", Namespace: metav1.NamespaceDefault}},
+				&bactchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: "j2", Namespace: metav1.NamespaceDefault}},
+			},
+			gvk: schema.GroupVersionKind{Group: "batch", Version: "v1", Kind: "Job"},
+			wantWorkloads: []v1beta1.Workload{
+				*testingworkload.MakeWorkload("wl1", metav1.NamespaceDefault).OwnerReference(jobGVK, "j1", "").Obj(),
+			},
+			wantJobList: &bactchv1.JobList{
+				TypeMeta: metav1.TypeMeta{Kind: "JobList", APIVersion: "batch/v1"},
+				Items: []bactchv1.Job{
+					{
+						TypeMeta:   metav1.TypeMeta{Kind: "Job", APIVersion: "batch/v1"},
+						ObjectMeta: metav1.ObjectMeta{Name: "j1", Namespace: metav1.NamespaceDefault},
+					},
+					{
+						TypeMeta:   metav1.TypeMeta{Kind: "Job", APIVersion: "batch/v1"},
+						ObjectMeta: metav1.ObjectMeta{Name: "j2", Namespace: metav1.NamespaceDefault},
+					},
+				},
```

Comment:
is this needed?


the same goes for most of the tests in this file

Reply:
But I agree that it's not need for current case.